frontend/src/metabase/metadata/pages/DataModel/components/PreviewSection/TablePreview.tsx

```diff
@@ -57,16 +60,33 @@ const TablePreviewBase = (props: Props) => {
   );
 };
 
-function useDataSample({ databaseId, field, fieldId, tableId }: Props) {
-  const datasetQuery = getPreviewQuery(databaseId, tableId, fieldId);
-
-  const { data, refetch, ...rest } = useGetAdhocQueryQuery({
-    ...datasetQuery,
-    _refetchDeps: field,
-  });
+function useDataSample({
+  databaseId,
+  field,
+  fieldId,
+  pkFields,
+  tableId,
+}: Props) {
+  const datasetQuery = getPreviewQuery(databaseId, tableId, fieldId, pkFields);
 
+  const { data, refetch, ...rest } = useGetAdhocQueryQuery(
+    {
+      ...datasetQuery,
+      ignore_error: true,
+      _refetchDeps: field,
+    },
+    {},
+  );
   const base = { ...rest, error: undefined, rawSeries: undefined };
 
+  if (rest?.status === "rejected") {
```

Comment:
Should this look at the `reason` of the rejection too to be more specific?

Reply:
I added 403 status check: 330b587866fbbf0080710ddcb2370adced63dbd9